experiments/nissl-to-ccfv3.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python
+# Copyright 2021, Blue Brain Project, EPFL
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script that maps Nissl to CCFv3."""
+from __future__ import annotations
+
+import argparse
+import logging
+import pathlib
+import sys
+from pathlib import Path
+
+import numpy as np
+from utils import get_results_dir
+
+from atlannot import load_volume
+from atlannot.ants import register, transform
+
+# Initialize the logger
+logger = logging.getLogger("nissl-to-ccfv3")
+DATA_FOLDER = pathlib.Path(__file__).resolve().parent.parent / "data"
+
+
+def parse_args():
+    """Parse arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--nissl_path",
+        type=Path,
+        default=DATA_FOLDER / "ara_nissl_25.nrrd",
+        help="""\
+        Path to Nissl Volume.
+        """,
+    )
+    parser.add_argument(
+        "--ccfv2_path",
+        type=Path,
+        default=DATA_FOLDER / "ccfv2_merged.nrrd",
+        help="""\
+        Path to CCFv2 annotation volume.
+        """,
+    )
+    parser.add_argument(
+        "--ccfv3_path",
+        type=Path,
+        default=DATA_FOLDER / "ccfv3_merged.nrrd",
+        help="""\
+        Path to CCFv3 annotation volume.
+        """,
+    )
+    return parser.parse_args()
+
+
+def check_and_load(path: pathlib.Path | str) -> np.array:
+    """Load volume if path exists.
+
+    Parameters
+    ----------
+    path
+        File path.
+
+    Returns
+    -------
+    volume : np.array
+        Loaded volume.
+    """
+    if not path.exists():
+        logger.error(f"The specified path {path} does not exist.")
+        return 1
+    volume = load_volume(path, normalize=False)
+    return volume.astype(np.float32)
+
+
+def slice_registration(
+    fixed: np.array, moving: np.array, nissl_slice: np.array | None = None
+) -> tuple[np.array, np.array | None]:
+    """Compute registration transform between a couple of slices.
+
+    Parameters
+    ----------
+    fixed
+        Fixed slice of type annotation.
+    moving
+        Moving slice ot type annotation.
+    nissl_slice
+        If needed, intensity slice in the same coordinate system as moving.
```

Comment:
IMO including the shapes of the arrays here would make it more readable. In general, it is a bit challenging to guess the shapes for the first time reader.

Reply:
I agree. Done in [2f1b58d](https://github.com/BlueBrain/atlas-annotation/pull/40/commits/2f1b58d5ebb5ed85f1453b1076bfc378ef2a1d70).